element_interface/intanloader/rhsutilities.py

```diff
@@ -0,0 +1,1037 @@
+        return ""
+    else:
+        return "s"
+
+
+# Define read_qstring function
+def read_qstring(fid):
+    """Read Qt style QString.
+
+    The first 32-bit unsigned number indicates the length of the string (in bytes).
+    If this number equals 0xFFFFFFFF, the string is null.
+
+    Strings are stored as unicode.
+    """
+
+    (length,) = struct.unpack("<I", fid.read(4))
+    if length == int("ffffffff", 16):
+        return ""
+
+    if length > (os.fstat(fid.fileno()).st_size - fid.tell() + 1):
+        print(length)
+        raise Exception("Length too long.")
+
+    # convert length from bytes to 16-bit Unicode words
+    length = int(length / 2)
+
+    data = []
+    for i in range(0, length):
+        (c,) = struct.unpack("<H", fid.read(2))
+        data.append(c)
+
+    if sys.version_info >= (3, 0):
+        a = "".join([chr(c) for c in data])
+    else:
+        a = "".join([unichr(c) for c in data])
+
+    return a
+
+
+# Define read_header function
+def read_header(fid):
+    """Reads the Intan File Format header from the given file."""
+
+    # Check 'magic number' at beginning of file to make sure this is an Intan
+    # Technologies RHD2000 data file.
+    (magic_number,) = struct.unpack("<I", fid.read(4))
+    if magic_number != int("0xD69127AC", 16):
+        raise Exception("Unrecognized file type.")
+
+    header = {}
+    # Read version number.
+    version = {}
+    (version["major"], version["minor"]) = struct.unpack("<hh", fid.read(4))
+    header["version"] = version
+
+    print("")
+    print(
+        "Reading Intan Technologies RHS2000 Data File, Version {}.{}".format(
+            version["major"], version["minor"]
+        )
+    )
+    print("")
+
+    # Read information of sampling rate and amplifier frequency settings.
+    (header["sample_rate"],) = struct.unpack("<f", fid.read(4))
+    (
+        header["dsp_enabled"],
+        header["actual_dsp_cutoff_frequency"],
+        header["actual_lower_bandwidth"],
+        header["actual_lower_settle_bandwidth"],
+        header["actual_upper_bandwidth"],
+        header["desired_dsp_cutoff_frequency"],
+        header["desired_lower_bandwidth"],
+        header["desired_lower_settle_bandwidth"],
+        header["desired_upper_bandwidth"],
+    ) = struct.unpack("<hffffffff", fid.read(34))
+
+    # This tells us if a software 50/60 Hz notch filter was enabled during
+    # the data acquisition.
+    (notch_filter_mode,) = struct.unpack("<h", fid.read(2))
+    header["notch_filter_frequency"] = 0
+    if notch_filter_mode == 1:
+        header["notch_filter_frequency"] = 50
+    elif notch_filter_mode == 2:
+        header["notch_filter_frequency"] = 60
+
+    (
+        header["desired_impedance_test_frequency"],
+        header["actual_impedance_test_frequency"],
+    ) = struct.unpack("<ff", fid.read(8))
+    (header["amp_settle_mode"], header["charge_recovery_mode"]) = struct.unpack(
+        "<hh", fid.read(4)
+    )
+
+    frequency_parameters = {}
```

Comment:
I find this hard to read, and I would expect it would be tough for newcomers to audit/maintain as a result. I'd suggest adopting the kinds of formatting adjustments I proposed in #74

Reply:
I think the best reason *not* to make such edits would be if we anticipated further developments from the original intan team. If the original repo is stable and unlikely to update, I would prioritize readability over exactness in the copy